src/autogpt_plugins/scenex/__init__.py

```diff
@@ -18,24 +21,25 @@ class AutoGPTSceneXPlugin(AutoGPTPluginTemplate):
 
     def __init__(self):
         super().__init__()
-        self._name = "Auto-GPT-Scenex-Plugin"
+        self._name = "ImageExplainer"
         self._version = "0.0.1"
-        self._description = "Auto-GPT Scenex Plugin: Describe any image by URL."
-
-    def post_prompt(self, prompt: PromptGenerator) -> PromptGenerator:
-        from .scenex_plugin.scenex_plugin import (
-            describe_image,
-            is_api_key_set,
+        self._description = (
+            "An Image Captioning Tool: Use this tool to generate a detailed caption for an image. "
+            "The input can be an image file of any format, and "
+            "the output will be a text description that covers every detail of the image."
         )
+        self.scenexplain = SceneXplain(os.getenv("SCENEX_API_KEY"))
+        self._is_api_key_set = bool(os.getenv("SCENEX_API_KEY"))
```

Comment:
nit: you could optimize this to not check twice registering the value within the class